src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -394,60 +432,100 @@ bool AdaptDispatch::CursorRestoreState()
 // - True.
 bool AdaptDispatch::CursorVisibility(const bool fIsVisible)
 {
-    // This uses a private API instead of the public one, because the public API
-    //      will set the cursor shape back to legacy.
-    _pConApi->SetCursorVisibility(fIsVisible);
+    _pConApi->GetTextBuffer().GetCursor().SetIsVisible(fIsVisible);
```

Comment:
Might be good to ruminate on what happens if we ever make the Cursor a "global good" (since we learned that an alt buffer and main buffer share cursor properties, for example -- does it make sense to have one per text buffer then?)

Reply:
The problem is that the cursor coordinates are absolute, so when the buffers aren't the same size, the coordinates won't be the same. And when it comes to the console buffers, I think their cursor positions might be completely independent - I can't remember now.

Cursor properties (like visibility, shape, blinking status) could possibly be made global though. I've made a few attempts at that in the past, but always got stuck on something. I haven't given up on the idea, but it wasn't as straightforward as I expected it to be.

Reply:
That totally makes sense. I legitimately forgot that the cursor's _position_ wasn't shared.